sway/commands/output/adaptive_sync.c

```diff
@@ -9,13 +10,25 @@ struct cmd_results *output_cmd_adaptive_sync(int argc, char **argv) {
 	if (argc == 0) {
 		return cmd_results_new(CMD_INVALID, "Missing adaptive_sync argument");
 	}
-
-	if (parse_boolean(argv[0], true)) {
-		config->handler_context.output_config->adaptive_sync = 1;
-	} else {
-		config->handler_context.output_config->adaptive_sync = 0;
+	
+	int current_value = 1;
+	
+	if(strcmp(argv[0], "toggle") == 0) {
+		struct output_config *oc = config->handler_context.output_config;
+		struct sway_output *sway_output = all_output_by_name_or_id(oc->name);
+		if (sway_output == NULL) {
+			return cmd_results_new(CMD_FAILURE,
+				"Cannot apply toggle to unknown output %s", oc->name);
+		}
+		oc = find_output_config(sway_output);
```

Comment:
Instead of using output configs, the current setting can be read off of `sway_output->wlr_output->adaptive_sync_status`.